packages/labs/analyzer/src/lib/utils.ts

```diff
@@ -47,6 +47,8 @@ export const hasProtectedModifier = (ts: TypeScript, node: ts.HasModifiers) => {
 const isPrivate = (ts: TypeScript, node: ts.Node) => {
   return (
     (ts.canHaveModifiers(node) && hasPrivateModifier(ts, node)) ||
+    ((ts.isPropertyDeclaration(node) || ts.isMethodDeclaration(node)) &&
+      ts.isPrivateIdentifier(node.name)) ||
```

Comment:
Does `ts.isPrivateIdentifier(()` need to be guarded by the property and method declaration checks?

Just thinking that this condition has a lot of clauses now, and it'd be easier to read if it were just:

```ts
return (
  hasPrivateModifier(ts, node) ||
  ts.isPrivateIdentifier(node.name) ||
  hasJSDocTag(ts, node, 'private')
);
```

If the guards are needed, maybe some variables would help reading:
```ts
const isTsPrivate = ts.canHaveModifiers(node) && hasPrivateModifier(ts, node));
const isMember = ts.isPropertyDeclaration(node) || ts.isMethodDeclaration(node);
const isJsPrivate = isMember && ts.isPrivateIdentifier(node.name);
const isJsDocPrivate = isMember && hasJSDocTag(ts, node, 'private');

return isTsPrivate || isJsPrivate || isJsDocPrivate;
```

If you don't want to refactor this, no action needed though. We have tests :)

Reply:
yeah i wasn't super jazzed about that, but at a certain point my desire to fight the compiler just drops off sharply, and if I can fill the gaps in the typescript ast's api with more checks - :shrug: good enough for now